src/utils/interpolatePixel.ts

```diff
@@ -61,8 +61,8 @@ function interpolateNearest(
   channel: number,
   interpolateBorder: BorderInterpolationFunction,
 ): number {
-  column = Math.round(column);
-  row = Math.round(row);
+  column = Math.floor(column);
+  row = Math.floor(row);
```

Comment:
Intuitively I'd say that the solution shouldn't involve changing the interpolation functions, because they are validated by other algorithms like rotate and transform.
Resize has the same issue with all interpolations which makes me think the problem is located only in the resize.ts file.

Reply:
I agree, I am currently reworking to this part.
I rename actuals `interpolateX` functions to `interpolateNeighbourX`. And provide `interpolatePositionX` functions.

So this algorithm change will be applied only on `nearest` interpolation. And will avoid to recompute `ny` on each `row` iteration